packages/keyring-controller/src/KeyringController.test.ts

```diff
@@ -2989,6 +3113,34 @@ describe('KeyringController', () => {
       });
     });
 
+    describe('withKeyring', () => {
+      it('should call withKeyring', async () => {
+        await withController(async ({ controller, messenger }) => {
+          const selector = { type: KeyringTypes.hd };
+          const fn = async () => 'foo';
+
+          await messenger.call('KeyringController:withKeyring', selector, fn);
+
+          expect(controller.withKeyring).toHaveBeenCalledWith(selector, fn);
+        });
+      });
+
+      it('should return the function result', async () => {
+        await withController(async ({ messenger }) => {
+          const selector = { type: KeyringTypes.hd };
+          const fn = async () => 'foo';
+
+          const result: string = await messenger.call(
+            'KeyringController:withKeyring',
+            selector,
+            fn,
+          );
```

Comment:
`Type 'unknown' is not assignable to type 'string'.`

Hmm, it looks like TS is not able to infer the return type coming from the handler's generic `<T>` when used with the messenger. This is not the case when calling the function `withKeyring` directly